### PR TITLE
PFM-ISSUE-9113: Move combineJavascripts task from build plugin to cplace-asc

### PR DESCRIPTION
**Pull request by vladimir-arsov**
Resolves [PFM-ISSUE-9113](https://base.cplace.io/pages/xmy7t4tn4jtherrdgptlwrad1/PFM-ISSUE-9113-Move-combineJavascripts-task-from-build-plugin-to-cplace-asc)

### CODE REVIEW

File: src/compiler/CombineJavascriptsCompiler.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import { CompilationResult, ICompiler } from './interfaces';
+import { cerr, debug, formatDuration, GREEN_CHECK } from '../utils';
+import * as fs from 'fs';
+import rimraf = require('rimraf');
+import * as webpack from 'webpack';
+import { Configuration } from 'webpack';
+
+export class CombineJavascriptsCompiler implements ICompiler {
+    public static readonly OUTPUT_DIR = '_generated_';
+    public static readonly ENTRY_FILE_NAME = 'javaScriptIncludesToBeCompressed.txt';
```
Comment (slaven3kopic) [inline src/compiler/CombineJavascriptsCompiler.ts:12]: Why is this file still needed after we publish all the assets to jFrog? There should be only 2 JS files left...`vendors.js` and `tsc.js`. And if we want we can combine them with this compiler

Comment (vladimir-arsov) [reply src/compiler/CombineJavascriptsCompiler.ts:12]: We cannot remove it yet since most of the repositories still use this file for the imports, instead of an `index.ts` file.

Comment (slaven3kopic) [reply src/compiler/CombineJavascriptsCompiler.ts:12]: Yeah, true. Do we actually want to compress everything into one single JS file? Maybe it makes more sense to split it ([Code Splitting](https://webpack.js.org/guides/code-splitting/)). The cplace load time could be faster.

Comment (vladimir-arsov) [reply src/compiler/CombineJavascriptsCompiler.ts:12]: This is how cplace is working. It expects one file `compressed.js` per plugin. We could look into code splitting later, as it could turn out to be a bigger task.

File: src/compiler/CombineJavascriptsCompiler.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            externals: {
+                jquery: 'jQuery'
+            },
+            resolveLoader: {
+                modules: [path.resolve(__dirname, '../../', 'node_modules')],
+            },
+            entry: {
+                compressed: this.getCombineJavascriptSources()
+            },
+            module: {
```
Comment (slaven3kopic) [inline src/compiler/CombineJavascriptsCompiler.ts:105]: Probably we should add some optimizations as well too keep the size of the file small. Something like [this](https://github.com/collaborationFactory/cplace-asc/blob/improvement/PFM-ISSUE-9113-move-combineJavascripts-task-from-build-plugin-to-cplace-asc/src/compiler/VendorCompiler.ts#L247)

**Review by slaven3kopic — APPROVED** [review]
